src/librustc_mir_build/hair/pattern/const_to_pat.rs

```diff
@@ -100,6 +105,11 @@ impl<'a, 'tcx> ConstToPat<'a, 'tcx> {
                 "search_for_structural_match_violation cv.ty: {:?} returned: {:?}",
                 cv.ty, structural
             );
+
+            if structural.is_none() && mir_structural_match_violation {
+                warn!("MIR const-checker found novel structural match violation");
```

Comment:
Should this be an ICE (i.e. `bug!`) instead of a `warn!` ?

Reply:
I vaguely recall a scenario where this occurred, although I forgot the specifics long ago. I'll see what making this ICE does to the test suite.

Reply:
It's actually the reverse. This one should (I think) never happen and the one below occurs regularly. This is because we override the result `search_for_structural_match_violation` when we know that the constant is actually an unqualified variant.